### PR TITLE
[DOC] Example of user-defined linestyle (TikZ linestyle)

### PR DESCRIPTION
**Pull request by rougier**
This is an example showing user-defined linestyle copying name and style of TikZ.

linestyles

### CODE REVIEW

File: examples/lines_bars_and_markers/linestyles.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ax.set_ylim(-0.5, len(linestyles)-0.5)
+plt.yticks(np.arange(len(linestyles)), linestyles.keys())
+plt.xticks([])
+
+for i, (name, linestyle) in enumerate(linestyles.items()):
+    ax.text(-0.5, i-0.4, str(linestyle), fontsize=8, ha="right",
+            color="blue", family="monospace")
+
+plt.tight_layout()
+# plt.savefig("linestyles.pdf")
```
Comment (NelleV) [inline examples/lines_bars_and_markers/linestyles.py]: I'd remove those two lines, thought I honestly don't have a strong feeling about it.

**Comment by NelleV** [comment]
Thanks again for a very nice addition to the gallery.

**Comment by rougier** [comment]
This could be the base for new linestyles maybe:

Name | Linestyle | Value
------|----------|-----
| `-`              | Solid | `(0, ())`
|`.:dense`   | Densely dotted | `(0, (1, 1))`
|`.:normal` or `.` | Dotted         | `(0, (1, 5))`
|`.:loose`   | Loosely dotted | `(0, (1, 10))`
|`--:dense`   | Densely dashed | `(0, (5, 1))`
|`--:normal` or `--` | Dashed         | `(0, (5, 5))`
|`--:loose`   | Loosely dashed | `(0, (5, 10))`
|`-.:dense`   | Densely dashdotted | `(0, (3, 1, 1, 1))`
|`-.:normal` or `-.` | Dashdotted         | `(0, (3, 5, 1, 5))`
|`-.:loose`   | Loosely dashdotted | `(0, (3, 10, 1, 10))`
|`-..:dense`   | Densely Dashdotdotted | `(0, (3, 1, 1, 1, 1, 1))`
|`-..:normal` or `-..` | Dashdotdotted         | `(0, (3, 5, 1, 5, 1, 5))`
|`-..:loose`   | Loosely Dashdotdotted | `(0, (3, 10, 1, 10, 1, 10))`

**Comment by afvincent** [comment]
I mention this just in case: there was some discussion related to line style patterns in #6547 a while ago, and the values that were chosen for the '-', ':' and '-.' line styles are a bit different.

Personally, I think it may be a good idea to add some extra default line styles like this PR suggest. Quite often, I wish there was (at least) a "dashdotdotted" style available.

About the example, I would have 2 remarks:
* it might be confusing to use the names "dotted", "dashdotted" and "dashed" for styles that do not use the default values.
* is it necessary to use `plt.xticks`and `plt.yticks` instead of the OO methods `ax.set_xticks` and `ax.set_yticks`?

Apart from these, I find the example to be very nice, as it may provide to the users some ready-to-use values for line style patterns, without having them to try a bunch of values on their own (like I often did...)

**Comment by rougier** [comment]
@afvincent Thanks for the review.

1. The names come from TikZ actually.
2. No reason to use `plt.xticks`, I need to change that.
3. Dotted /dashed: I didn't know what are the default values so I tried to guess them

Also thanks for the link to the PR. Do you now if there's a way to have round caps in 2.0 ?

**Comment by rougier** [comment]
Actually it doesn't work with `ax.set_yaxis()`

**Comment by tacaswell** [comment]
`Line2D.set_dash_capstyle`   

http://matplotlib.org/api/lines_api.html#module-matplotlib.lines

There is also a `dash_jointstyle`

**Comment by rougier** [comment]
@tacaswell Thanks. I would (personally) prefer round caps by defaults but I imagine it has been already discussed.

**Comment by rougier** [comment]
Is there anything else to do with this PR ?

**Comment by tacaswell** [comment]
backported to v2.x as 7ed77e4

**Pull request merged**